umbral/umbral.py

```diff
@@ -285,19 +291,19 @@ def reconstruct_capsule(self, cFrags):
                 v = v + (cFrag.v1 * lambda_i)
 
             return ReconstructedCapsule(e_prime=e, v_prime=v, x=cFrag_0.point_eph_ni)
-
         else: #if len(reencrypted_keys) == 1:
             return ReconstructedCapsule(e_prime=cFrag_0.e1, v_prime=cFrag_0.v1, x=cFrag_0.point_eph_ni)
 
-    def decapsulate_reencrypted(self, pub_key, priv_key, ctxt_combined, orig_pk, orig_ciphertext, key_length=32):
+    def decapsulate_reencrypted(self, pub_key: Point, priv_key: BigNum, capsule, orig_pub_key: Point,
+                                orig_ciphertext: Capsule, key_length=32):
```

Comment:
So, @cygnusv and @tuxxy, as I look toward implementing this stuff in the KMS, this is the epicenter of my wonder so far.

I notice (and have pointed out here via type hints) that public keys are `Points`, while private keys are `BigNums`.  

How do you propose working that into the KMS codebase, where we have classes for these things?  Shall we start using `Point` and `BigNum` as the basis for keys up there as well?